service/chartconfig/chart_framework.go

```diff
@@ -42,6 +43,9 @@ func NewChartFramework(config ChartFrameworkConfig) (*framework.Framework, error
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
 	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.ProjectName must not be empty")
+	}
```

Comment:
This check was missing. There is no check for WatchNamespace because "" is a valid value.